projects/ad469x_fmcz/src/common/common_data.c

```diff
@@ -0,0 +1,152 @@
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "common_data.h"
+#include "no_os_pwm.h"
+#include "no_os_spi.h"
+#include "no_os_uart.h"
+#include "ad469x.h"
+
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+struct no_os_uart_init_param iio_uart_ip = {
+	.device_id = UART_DEVICE_ID,
+	.irq_id = UART_IRQ_ID,
+	.asynchronous_rx = true,
+	.baud_rate = UART_BAUDRATE,
+	.size = NO_OS_UART_CS_8,
+	.parity = NO_OS_UART_PAR_NO,
+	.stop = NO_OS_UART_STOP_1_BIT,
+	.extra = UART_EXTRA,
+	.platform_ops = UART_OPS,
+};
+
+struct no_os_spi_init_param spi_init = {
+	.chip_select = SPI_CS,
+	.max_speed_hz = 80000000,
+	.mode = NO_OS_SPI_MODE_3,
+	.platform_ops = SPI_OPS,
+	.extra = (void*)SPI_EXTRA,
+};
+struct spi_engine_offload_init_param spi_engine_offload_init_param = {
+	.offload_config = OFFLOAD_RX_EN,
+	.rx_dma_baseaddr = DMA_BASEADDR,
+};
+
+struct axi_clkgen_init clkgen_init = {
+	.name = "rx_clkgen",
+	.base = RX_CLKGEN_BASEADDR,
+	.parent_rate = 100000000,
+};
+
+struct no_os_pwm_init_param trigger_pwm_init = {
+	.period_ns = 1000,
+	.duty_cycle_ns = 10,
+	.polarity = NO_OS_PWM_POLARITY_HIGH,
+	.platform_ops = PWM_OPS,
+	.extra = PWM_EXTRA,
+};
+
+struct no_os_gpio_init_param ad469x_resetn = {
+	.number = GPIO_RESETN_1,
+	.platform_ops = GPIO_OPS,
+	.extra = GPIO_EXTRA,
+};
+
+struct ad469x_init_param ad469x_init_param = {
+#ifdef AD4695_DEV
+	.dev_id = ID_AD4695,
+#elif defined(AD4696_DEV)
+	.dev_id = ID_AD4696,
+#elif defined(AD4697_DEV)
+	.dev_id = ID_AD4697,
+#elif defined(AD4698_DEV)
+	.dev_id = ID_AD4698,
+#endif
+#if (defined(AD4695_DEV) || defined(AD4696_DEV))
+	.max_data_channels = 16,
```

Comment:
ill add a driver structure with per chip max_channel.
and this will be just a optional num_channel. if not set max channel will be used.